webapp/views.py

```diff
@@ -495,7 +540,37 @@ def advantage_view():
                     subscription["purchasedProductListings"]
                 )
 
+            for subscription in yearly_subscriptions:
+                for purchase in subscription["purchasedProductListings"]:
+                    listing_id = purchase["productListing"]["id"]
+                    purchased_product_id = listings_to_products[listing_id]
+                    yearly_products[purchased_product_id][
+                        "quantity"
+                    ] = purchase["value"]
+
             for contract in account["contracts"]:
+                contract["period"] = get_contract_period(contract)
+
+                period_products = (
+                    monthly_products
+                    if contract["period"] == "yearly"
+                    else yearly_products
+                )
+
+                product_name = contract["contractInfo"]["products"][0]
+                contract_product = period_products[product_name]
+                contract["is_cancelable"] = False
```

Comment:
`is_cancellable` could be determined by a single lookup:
`monthly_subscriptions[contrac.products[0]]`. If it exists, then we have a monthly listing to cancel on that contract.